### ISSUE TITLE
Converter org.modelmapper.internal.converter.DateConverter@6765246d failed to convert java.lang.String to java.util.Date

### ISSUE THREAD

**Issue by Sam-Dragon** [modelmapper/modelmapper#628]
Hi,

Please find the required details

Java Version - 11
Model Mapper Version - 2.3.8

Above issue is coming after we deploy the application

Steps
1. During the first deployment when We make changes in application & deploy - first time we are getting above error
2. During the second deployment when we deploy without making changes - Error disappears

We are seeing this strange behavior. Please help

Thank you

**Comment by chhsiao90** [comment]
Do you have stacktrace of the exception? Do you have reproducible code? Thanks.

**Comment by Sam-Dragon** [comment]
> Do you have stacktrace of the exception? Do you have reproducible code? Thanks.
- Hi Chub-Han, I have found the solution to the problem. I will sharing the details for the same

**Comment by Sam-Dragon** [comment]
After lot of research, I found the solution to the issue that it was not w.r.t deployment.

The issue was because of model mapper configuration of MATCHING STRATEGIES. It happened because my code updating the MATCHING STRATEGIES with in same application

So let me explain in detail, I was using spring boot application

In main class of spring boot app, MATCHING STRATEGY of model mapper was set to STRICT which applies to all the services where model mapper is used

In one of my service, I was updating MATCHING STRATEGY of model mapper was set to LOOSE which actually applies to all service

Basically once the MATCHING STRATEGY moves from STRICT to LOOSE, it will never become STRICT & there the problem starts where services will start breaking

Services will work well once the application is re-started again

So, guys ensure whenever you are using model mapper, set only one MATCHING STRATEGY per application.

Thank you
